Remove X-Download-Options default header

sabljak opened the pull request:
### Summary

<!-- Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together. -->

Since `X-Download-Options` header is only used by the soon deprecated Internet Explorer to prevent phishing attacks. 
It makes sense to remove this header as a default one.

This PR solves [#43948](https://github.com/rails/rails/issues/43948)

Added new header defaults as a framework default from Rails 7.1.

<!--  ### Other Information -->

<!-- If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information. -->

ghiculescu commented on railties/CHANGELOG.md:
```diff
@@ -1,3 +1,11 @@
+    will be discontinued in 2022 and since Rails 7 does not fully
+    support Internet Explorer this header should not be a default one.
+
+    This change is a framework default from Rails 7.1.
```
This line isn't needed, it's implicit from which part of the changelog it's in.

sabljak replied:
Alright!

ghiculescu commented on railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_1.rb.tt:
```diff
@@ -8,3 +8,12 @@
 # Read the Guide for Upgrading Ruby on Rails for more info on each option.
 # https://guides.rubyonrails.org/upgrading_ruby_on_rails.html
+
+# Remove the default X-Download-Options headers since it is used only by Internet Explorer
```
```suggestion
# Remove the default X-Download-Options headers since it is used only by Internet Explorer.
# If you need to support Internet Explorer, add back `"X-Download-Options" => "noopen"`.
```

sabljak commented:
Thanks for the review!

I've made the requested changes.

sabljak commented:
Hey @ghiculescu 

Is there anything else I should add to this PR?

ghiculescu commented:
It looks good to me. But it needs review from someone in the core team. They are pretty busy but they will get to it (but in the new year is more likely).

rafaelfranca commented on railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_1.rb.tt:
```diff
@@ -8,3 +8,13 @@
+#   "X-Content-Type-Options" => "nosniff",
+#   "X-Permitted-Cross-Domain-Policies" => "none",
+#   "Referrer-Policy" => "strict-origin-when-cross-origin"
+# }
```
Missing line break here. Can you add it?

sabljak replied:
Sure. Added!

sabljak committed:
Remove X-Download-Options default header

Since X-Download-Options header is only used by the soon deprecated Internet Explorer, it makes sense to remove this header as a default one.
